utils/simulator/simulator.py

```diff
@@ -0,0 +1,198 @@
+            logs_task.cancel()
+            await process.communicate()
+
+    @staticmethod
+    def default_simulator_path() -> Optional[Path]:
+        mini404_dep_path = get_dependency_directory('mini404')
+        simulator_path = mini404_dep_path / 'qemu-system-buddy'
+        if simulator_path.exists():
+            return simulator_path
+        else:
+            return None
+
+    async def command(self, command: str, readline=False):
+        async def issue_command():
+            assert self.scriptio_writer, 'scriptio socket isn\'t connected'
+            self.scriptio_writer.write(command.encode('utf-8') + b'\n')
+
+        async def wait_for_script_finished_line():
+            async for line in self.logs:
+                if line.strip() == 'ScriptHost: Script FINISHED':
+                    break
+
+        await asyncio.gather(issue_command(), wait_for_script_finished_line())
+
+        if readline:
+            line = await self.scriptio_reader.readline()
+            return line.decode('utf-8').strip()
+
+    #
+    # encoder primitives
+    #
+
+    async def encoder_click(self):
+        await self.command('encoder-input::Push()')
+
+    async def encoder_push(self):
+        raise NotImplementedError()
+
+    async def encoder_release(self):
+        raise NotImplementedError()
+
+    async def encoder_rotate_left(self):
+        await self.command(f'encoder-input::Twist(1)')
+
+    async def encoder_rotate_right(self):
+        await self.command(f'encoder-input::Twist(1)')
+
+    #
+    # screen primitives
+    #
+
+    async def screen_take_screenshot(self) -> Image.Image:
+        screenshot_path = self.tmpdir / (str(uuid.uuid4()) + '.png')
+        await self.command(f'st7789v::Screenshot({screenshot_path})')
+        return Image.open(screenshot_path)
+
+    #
+    # temperature primitives
+    #
+
+    def _get_thermistor_idx(self, thermistor: Thermistor):
+        if self.machine == MachineType.MINI and thermistor == Thermistor.BED:
+            return 1
+        elif self.machine == MachineType.MINI and thermistor == Thermistor.NOZZLE:
+            return 2
+        raise NotImplementedError('dont know the index of thermistor %s on %s',
+                                  thermistor, self.machine)
+
+    async def temperature_set(self, thermistor: Thermistor,
+                              temperature: float):
+        thermistor_idx = self._get_thermistor_idx(thermistor)
+        await self.command(f'thermistor{thermistor_idx}::Set({temperature})')
+
+    async def temperature_get(self, thermistor: Thermistor) -> float:
+        thermistor_idx = self._get_thermistor_idx(thermistor)
+        temp_str = await self.command(f'thermistor{thermistor_idx}::GetTemp()',
+                                      readline=True)
+        return float(temp_str)  # type: ignore
+
+    #
+    # network primitives
+    #
+
+    def network_ip_address_get(self, interface: NetworkInterface) -> str:
+        raise NotImplementedError()
```

Comment:
This might actually be easier to just set to static in eeprom than to _get_ it later on.

Reply:
Actually, let's delete this for now. Originally this was something I worked on but then I decided to take another path...

Reply:
Deleted...